src/components/KSelect/KSelect.vue

```diff
@@ -600,6 +600,16 @@ export default defineComponent({
       }
     }
 
+    // If the select is readonly with chevron, need to add the bg-color to the parent container
+    const addReadonlyStyle = () => {
+      const all = document.querySelector('.k-select-input.no-filter.select-input-container .k-input[readonly]')as HTMLInputElement
```

Comment:
I don’t think this is a viable solution. If there are multiple KSelect elements on the page, this code would only ever style the first one. 

Also, I don’t think we want to resort to using JavaScript to apply styles